### PR TITLE
48877_supervisor_edit_trainee_fix

### PR DESCRIPTION
**Pull request by yasin-buet**

### CODE REVIEW

**Comment by asifbuetcse** [comment]
Ok

File: app/Http/Controllers/Supervisor/UsersController.php [hunk 1]
```diff
@@ -81,10 +81,12 @@ public function show($id)
      * @param  int  $id
      * @return \Illuminate\Http\Response
      */
-    public function edit($userId)
+    public function edit(Request $request, $userId)
     {
-        $supervisor = user::find($userId); 
-        $this->authorize('edit', $supervisor);        
+        $supervisor = user::find($userId);
+        if ($request->input('type') != 'trainee') {
```
Comment (dinhhoanglong91) [inline app/Http/Controllers/Supervisor/UsersController.php]: Are there any types without `trainee` ?

File: app/Http/Controllers/Supervisor/UsersController.php [hunk 2]
```diff
@@ -101,8 +103,7 @@ public function update(Request $request, $id)
         $this->validate($request, [
             'name' => 'required|max:255',
             'email' => 'required|email|max:255'
-        ]);  
-        $this->authorize('edit', $user);
```
Comment (dinhhoanglong91) [inline app/Http/Controllers/Supervisor/UsersController.php]: You don't authorize here, do you?

Comment (yasin-buet) [inline app/Http/Controllers/Supervisor/UsersController.php]: no, i don't authorize here.

File: app/Http/Controllers/Supervisor/UsersController.php (hunk 1 above)
Comment (yasin-buet) [inline app/Http/Controllers/Supervisor/UsersController.php]: there are two things, superviosr edit his own profile and supervisor edit trainee's profile. therefore checked the type trainee.

File: app/Http/Controllers/Supervisor/UsersController.php (hunk 2 above)
Comment (dinhhoanglong91) [inline app/Http/Controllers/Supervisor/UsersController.php]: If you don't authorize and other people send request to this page without accessing `edit` page, how do you do?

**Comment by dinhhoanglong91** [comment]
ok

**Pull request merged**